src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnUserAccountService.java

```diff
@@ -0,0 +1,167 @@
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount findByProviderAndSubject(String provider, String subject) throws NoSuchUserException {
+        WebAuthnUserAccount account = accountRepository.findByProviderAndSubject(provider, subject);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount findByCredentialId(ByteArray credentialId) {
+        WebAuthnCredential cred = credentialRepository.findByCredentialId(credentialId.getBase64());
+        if (cred == null) {
+            return null;
+        }
+        return accountRepository.detach(cred.getParentAccount());
+    }
+
+    /*
+     * CRUD
+     */
+    public WebAuthnUserAccount addAccount(
+            WebAuthnUserAccount reg) throws RegistrationException {
+
+        try {
+            // we explode model
+            WebAuthnUserAccount account = new WebAuthnUserAccount();
+
+            account.setCredentials(reg.getCredentials());
+            account.setEmailAddress(reg.getEmailAddress());
+            account.setProvider(reg.getProvider());
+            account.setRealm(reg.getRealm());
+            account.setUserHandle(reg.getUserHandle());
+            account.setUsername(reg.getUsername());
+
+            account = accountRepository.saveAndFlush(account);
+            account = accountRepository.detach(account);
+
+            return account;
+        } catch (Exception e) {
+            throw new RegistrationException(e.getMessage());
+        }
+    }
+
+    public WebAuthnUserAccount updateAccount(
+            long id,
+            WebAuthnUserAccount reg) throws NoSuchUserException, RegistrationException {
+        if (reg == null) {
+            throw new NoSuchUserException();
+        }
+
+        WebAuthnUserAccount account = accountRepository.findOne(id);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+
+        try {
+            // we explode model and update every field
+            account.setCredentials(reg.getCredentials());
+            account.setEmailAddress(reg.getEmailAddress());
+            account.setProvider(reg.getProvider());
+            account.setRealm(reg.getRealm());
+            account.setUserHandle(reg.getUserHandle());
+            account.setUsername(reg.getUsername());
+
+            account = accountRepository.saveAndFlush(account);
+            account = accountRepository.detach(account);
+
+            return account;
+        } catch (Exception e) {
+            throw new RegistrationException(e.getMessage());
+        }
+    }
+
+    public void deleteAccount(String provider, String subject) {
+        WebAuthnUserAccount account = accountRepository.findByProviderAndSubject(provider, subject);
+        accountRepository.delete(account);
```

Comment:
before removing the account you need to collect and remove all the associated credentials

Reply:
Fixed in 1087830eaa48c4faa46627f1b7c848eab27c8805